src/reducers/expressions.js

```diff
@@ -9,38 +9,59 @@ import type { Expression } from "../types";
 import type { Action } from "../actions/types";
 import type { Record } from "../utils/makeRecord";
 
+function expressionType(visible) {
+  return visible ? "visible" : "hidden";
+}
+
 type ExpressionState = {
-  expressions: I.List<Expression>,
+  expressions: I.Map<string, Expression>,
 };
 
 const State = makeRecord(
   ({
-    expressions: I.List(restoreExpressions()),
+    expressions: I.Map({
+      visible: I.List(restoreExpressions()),
+      hidden: I.List(),
+    }),
```

Comment:
nit: it might be simpler to keep one list and then filter the list when persisting and selecting values. It's *slightly* less efficient but will be easier when we want all expressions.